rustls-cert-gen/src/cert/mod.rs

```diff
@@ -0,0 +1,44 @@
+use rcgen::{Certificate, CertificateParams, Error};
+use std::{fs::File, io, path::Path};
+
+mod params;
+pub use params::BuildParams;
+
+mod ca;
+mod entity;
+mod signature;
+
+pub trait CertParams {
+	fn params(&self) -> &CertificateParams;
+}
+
+#[derive(Debug, Clone)]
+/// Struct to represent a Serialized Cert/Key pair
+pub struct SerializedEntity {
+	pub cert_pem: String,
+	pub key_pem: String,
+}
+
+impl SerializedEntity {
+	pub fn write(&self, dir: &Path, name: &str) -> Result<(), io::Error> {
+		use std::io::Write;
+
+		let key_path = dir.join(format!("{name}.key.pem"));
+		let mut key_out = File::create(key_path)?;
+		write!(key_out, "{}", &self.cert_pem)?;
+
+		let cert_path = dir.join(format!("{name}.pem"));
+		let mut cert_out = File::create(cert_path)?;
+		write!(cert_out, "{}", &self.cert_pem)?;
+
+		Ok(())
+	}
+}
+
+/// Methods common to all Certificate types
+pub trait Cert {
```

Comment:
I'm not sure this trait is worthwhile. For one thing, it creates an awkward situation where `serialize` has to take an `Option` for the signer even though the `Ca` implementation always throws the arg away.

It also doesn't seem like `main.rs` ever handles "Certs" generically, it either has a `Ca` or an `EndEntity` and if both implemented their own `serialize` functions they could be called as required without needing this indirection.